packages/dds/tree/api-report/tree.api.md

```diff
@@ -1797,8 +1797,8 @@ export const enum TreeNavigationResult {
 }
 
 // @public
-export abstract class TreeNode implements WithType {
-    abstract get [type](): string;
+export interface TreeNode extends WithType {
+    get [type](): string;
 }
 
```

Comment:
In order to leave the prototype chain unmodified, TreeNodes cannot inherit from a common base class.

I converted TreeNode to an interface instead and replaced the few internal 'instanceof TreeNode' checks with 'isTreeNode(..)'.